packages/react-native/ReactAndroid/src/main/java/com/facebook/react/views/image/ReactImageView.kt

```diff
@@ -439,6 +444,38 @@ public class ReactImageView(
       imagePipeline.evictFromCache(uri)
     }
 
+    if (cacheControl == ImageCacheControl.ONLY_IF_CACHED) {
+      val imagePipeline = Fresco.getImagePipeline()
+      val dataSource = imagePipeline.isInDiskCache(uri)
+
+      dataSource.subscribe(object : BaseDataSubscriber<Boolean>() {
+        override fun onNewResultImpl(dataSource: DataSource<Boolean>) {
+          if (!dataSource.isFinished()) return
+
+          val isInCache: Boolean = dataSource.getResult() ?: false
+
+          if (isInCache) {
+            setupImageRequest(uri, cacheControl, postprocessor, resizeOptions)
```

Comment:
If the image is not available in any of our caches, this should dispatch a load fail (i.e. `eventDispatcher.dispatchEvent(createErrorEvent(...))`. 

Alternatively, we could take advantage of Fresco's lowest permitted request levels (per https://frescolib.org/docs/image-requests.html). A lowest permitted level of `DISK_CACHE` should circumvent the need to check caches altogether. It's important to confirm that `onError` is called in JS for cases where the image is not available from cache.

Reply:
Thanks a lot for your quick review @Abbondanzo, I've added error handling in d01ccb83a3e20cb80241fd80840c0541f60d087c.

The alternative with the request levels seems interesting and could be a good optimisation. Will look further into it

Reply:
Brought this change over to the Fresco team and they recommended just taking advantage of the request level for a few reasons:
-  Under the hood, Fresco propagates the error from [here](https://github.com/facebook/fresco/blob/8e79c29fab817fe0cf789f46c7ca722416727482/imagepipeline/src/main/java/com/facebook/imagepipeline/producers/DecodeProducer.kt#L126) so we don't have to implement any special error dispatching. This error will automatically get passed into our downloadListener's `onFailure` method so there's no need to add custom branching logic for dispatching errors
- Reading from bitmap and encoded memory caches is synchronous and would yield the image result much earlier on. This would best be done before asking the disk cache
- Even though this change reads disk cache from a different thread, it doesn't create the image request until the disk cache read finishes. We should send the image request as soon as possible to let Fresco decide what decoder to read from
- This change reads from disk cache twice, first to detect if the image is available and then to actually decode it. We can avoid the second round trip by letting Fresco read from cache and perform decoding all at once

This should be as easy as tacking on `setLowestPermittedRequestLevel` like so:
```kotlin
 val requestLevel =
        if (cacheControl == ImageCacheControl.ONLY_FROM_CACHE) RequestLevel.FULL_FETCH
        else RequestLevel.DISK_CACHE

val imageRequestBuilder =
        ImageRequestBuilder.newBuilderWithSource(uri)
            // ... other setters ...
            .setLowestPermittedRequestLevel(requestLevel)
```

Reply:
Thanks a lot again for the detailed feedback and bringing it to the team. I've implemented this approach and it looks way simpler now!

Also retested and it gives the same loading output as with the other approach, and also the errors are being propagated to JS when there is no image available in our caches.